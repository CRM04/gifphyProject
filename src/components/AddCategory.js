import React, { useState } from 'react';
import propTypes from 'prop-types';


export const AddCategory = ({ setHeroes }) => {
    const [inputValue, setinputValue] = useState('');

    const inputValueChange = (e) => {
        setinputValue(e.target.value);
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        if (inputValue.trim().length > 1) {
            setHeroes(her => [inputValue, ...her]);
            setinputValue('');
        }
    }

    return (
        <form onSubmit={handleSubmit}>
            <p> {inputValue} </p>
            <input type="text" value={inputValue} onChange={inputValueChange} ></input>
        </form>
    );

}

AddCategory.propTypes = {
    setHeroes: propTypes.func.isRequired
}

