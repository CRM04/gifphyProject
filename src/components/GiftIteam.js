import React from 'react';
import propTypes from 'prop-types';

export const GiftItem = ({ title, url }) => {

    return (
        <div className="card">
            <img src={url} alt={title} ></img>
            <p> {title} </p>
        </div>
    );
}

GiftItem.prototypes = {
    title: propTypes.string.isRequired,
    url: propTypes.string.isRequired
}