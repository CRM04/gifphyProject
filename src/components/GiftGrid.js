import React from 'react';
import { useFetchGif } from '../hooks/useFetchGift';
import { GiftItem } from './GiftIteam';
import PropTypes from 'prop-types';

export const GiftGrid = ({ heroe }) => {

    const { data, loading } = useFetchGif(heroe);

    return (
        <>
            <h3 className="animate__animated animate__fadeIn"> {heroe} </h3>
            { loading && <p>Loading... </p> }
            <div className="card-grid animate__animated animate__fadeIn">
                {
                    data.map(gif => <GiftItem {...gif} key={gif.id} />)
                }
            </div>
        </>
    );
}

GiftGrid.propTypes = {
    heroe: PropTypes.string.isRequired
}