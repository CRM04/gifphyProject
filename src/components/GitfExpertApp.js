import React, { useState } from 'react';
import { AddCategory } from './AddCategory';
import { GiftGrid } from './GiftGrid';

const GiftExpertApp = ( {categories = []} ) => {

    const [Heroes, setHeroes] = useState(categories);

    return (
        <div>
            <h2> GitfExpertApp </h2>
            <AddCategory setHeroes={setHeroes} />
            <hr></hr>
            {Heroes.map(heroe => <GiftGrid heroe={heroe} key={heroe} />)}
        </div>
    );


}

export default GiftExpertApp;