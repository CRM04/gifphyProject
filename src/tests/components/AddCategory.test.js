import { shallow } from "enzyme";
import { AddCategory } from '../../components/AddCategory';
import '@testing-library/jest-dom';

describe('AddCategories test suit', () => {

    //this a custom function provided by jest 
    const setHeroes = jest.fn();
    let wrapper = shallow(<AddCategory setHeroes={setHeroes} />);


    beforeEach(() => {
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setHeroes={setHeroes} />);
    });


    test('AddCategory snapshot should match', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('input value should change', () => {
        const input = wrapper.find('input');
        const value = 'Hola mundo';
        input.simulate('change', { target: { value } });
        const pText = wrapper.find('p').text().trim();
        expect(pText).toBe(value);
    });


    test('form should not be submit', () => {
        wrapper.find('form').simulate('submit', { preventDefault() { } });
        expect(setHeroes).not.toHaveBeenCalled();
    });


    test('form should have been called', () => {
        const value = 'Hola';
        //simulates events
        wrapper.find('input').simulate('change', { target: { value } });
        wrapper.find('form').simulate('submit', { preventDefault() { } });
        //this test setHeroes have been called at least once, and with a function argument 
        expect(setHeroes).toHaveBeenCalled();
        expect(setHeroes).toHaveBeenCalledTimes(1);
        expect(setHeroes).toHaveBeenCalledWith(expect.any(Function));

        expect(wrapper.find('input').prop('value')).toBe('');
    })

});
