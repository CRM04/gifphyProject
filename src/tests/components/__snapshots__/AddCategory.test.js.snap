// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCategories test suit AddCategory snapshot should match 1`] = `
<form
  onSubmit={[Function]}
>
  <p>
     
     
  </p>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
